pkg/reconciler/knativeeventing/knativeeventing.go

```diff
@@ -246,26 +239,36 @@ func (r *Reconciler) delete(instance *eventingv1alpha1.KnativeEventing) error {
 	if len(instance.GetFinalizers()) == 0 || instance.GetFinalizers()[0] != finalizerName {
 		return nil
 	}
+
 	r.Logger.Info("Deleting resources")
-	var RBAC = mf.Any(role, rolebinding)
-	if len(r.eventings) == 0 {
-		// delete the deployments first
-		if err := r.config.Filter(mf.ByKind("Deployment")).Delete(); err != nil {
-			r.Logger.Warn(err, "Error deleting deployments")
+
+	// List all KnativeEventings to determine if cluster-scoped resources should be deleted.
+	kes, err := r.KnativeEventingClientSet.OperatorV1alpha1().KnativeEventings("").List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all KnativeEventings: %w", err)
+	}
+
+	if len(kes.Items) == 1 {
```

Comment:
Curious whether there's also a race condition here. Specifically, if the API server tells us there are 2 while both are being deleted, is it safe to rely on the other's reconcile loop to delete the cluster-scoped resources? I assume so, but just checking.

Reply:
Hm, you are right. I'll make this a little more bulletproof by checking if all items have a DeletionTimestamp set, i.e.: All resources are being deleted.